lib/NetworkManager.js

```diff
@@ -110,6 +111,21 @@ class NetworkManager extends EventEmitter {
     await this._updateProtocolRequestInterception();
   }
 
+  /**
+   * @param {!Array<string>} blockedUrls
+   */
+  async setBlockedURLs(blockedUrls) {
+    this._blockedUrls = [];
+
+    console.assert(Array.isArray(blockedUrls), `Expected blocked urls to be an array of Strings, but "${typeof blockedUrls}" is found.`);
+
+    blockedUrls.forEach(url => {
+      console.assert(helper.isString(url), `Expected url "${url}" to be a String, but "${typeof url}" is found`);
+      this._blockedUrls.push(url);
+    });
+    await this._client.send('Network.setBlockedURLs', { urls: this._blockedUrls });
```

Comment:
The `Network.setBlockedURLs` is deprecated; please don't use it.

There are interception patterns as a part of `setRequestInterceptionEnabled`. However, we already have request interception implemented so it should cover the usecase.

Why would you want this feature? Is it out of convenience?

Reply:
Seems we already have this, e.g. https://github.com/GoogleChrome/puppeteer/blob/master/examples/block-images.js

If you would make it that you can attach a handler to the `browser` element and that somehow attaches it to all sub-pages, so it intercepts every single request, that would be great. (Trying to add via `browser.on('targetcreated',handler)` seems to miss the first few network events.

Reply:
Apologies, I thought the `experimental` flag on the docs meant it was new rather than deprecated.

the TLRD; of why I want/need the feature is to block third party scripts when testing things as they slow things down. It would be nice to just set some blocked urls rather than have to intercept every request and do it manually.